### PR TITLE
added feature to insert key-value pair at given path in consul object

### PR DESCRIPTION
**Pull request by sam2426**
added feature to insert key-value pair at given path in consul object

### CODE REVIEW

File: src/migrator/QueryFactory.ts [hunk 1]
```diff
@@ -60,33 +60,35 @@ export class QueryFactory {
...
+        if(jp.query(parsed, this._jsonpath).length) {
+          jp.apply(
+            parsed,
+            this._jsonpath,
+            this._jsonpathFunc ?? (() => this._value)
+          )
+        } else {
+          helper.insertObjectAtJPath(this._jsonpath, this._value);
+        }
+        return consul.set(this.key, JSON.stringify(parsed, null, 2));
```
Comment (arketec) [inline src/migrator/QueryFactory.ts:79]: ` } else {
          helper.insertObjectAtJPath(this._jsonpath, this._value);
        }
        return consul.set(this.key, JSON.stringify(parsed, null, 2));`

should be:
 } else {
          helper.insertObjectAtJPath(this._jsonpath, this._value);
          return consul.set(this.key, helper.toString(true));
        }
        return consul.set(this.key, JSON.stringify(parsed, null, 2));

File: src/templates/migration.ts.ejs [hunk 2]
```diff
@@ -23,7 +23,17 @@
...
+    // Note: use jsonpath to add new key-value pair in consul objects
+    // await consul
+    //  .key(`Test_Key`)
+    //  .jsonpath(`$.b.f`)
+    //  .val({
+    //    "path": "www.test.com",
+    //    "method": "GET"
+    //  })
+    //  .save();
+    //
```
Comment (arketec) [inline src/templates/migration.ts.ejs:36]: Make a note here about this not supporting the full jsonpath spec. This functionality only supports the "." notation from jsonpath. see here for full: https://www.npmjs.com/package//jsonpath

**Comment by arketec** [comment]
@sam2426 
this change looks good overall, just go ahead and fix those notes I mentioned and I will merge this in

File: src/migrator/QueryFactory.ts (hunk 1 above)
Comment (sam2426) [reply src/migrator/QueryFactory.ts:80]: added.

File: src/templates/migration.ts.ejs (hunk 2 above)
Comment (sam2426) [reply src/templates/migration.ts.ejs:36]: added the comment in .ejs file

**Comment by sam2426** [comment]
added the suggested changes

**Pull request merged**